src/uncrustify_emscripten.cpp

```diff
@@ -355,6 +290,61 @@ string show_config()
 }
 
 
+std::vector<OptionGroup*> get_groups()
+{
+   std::vector<OptionGroup*> groups;
+   groups.reserve(5);
+
+   for(size_t i = 0; true; ++i)
+   {
+      OptionGroup* group = get_option_group(i);
+      if(!group)
+      {
+         break;
+      }
+
+      groups.push_back(group);
+   }
+
+   return groups;
+}
+
+
+std::vector<GenericOption*> get_options()
+{
+   std::vector<GenericOption*> options;
+   options.reserve(get_option_count());
+
+   for(size_t i = 0; true; ++i)
+   {
+      OptionGroup* group = get_option_group(i);
+      if(!group)
+      {
+         break;
+      }
+
+      options.insert(
+         end(options),
+         begin(group->options),
+         end(group->options)
+      );
+   }
+
+   return options;
+}
+
+
+//! resets all options to their default values
+void reset_options()
+{
+   auto options = get_options();
+   for(auto* option : options)
+   {
+      option->read(option->defaultStr().c_str());
```

Comment:
If this is the only reason for the change to `defaultStr`, there are surely better ways. An easy one would be to add a `reset()` method to `Option`. Possibly the only reason that doesn't already exist is because uncrustify proper didn't need it. (Such a method would also save an unnecessary conversion to and from a string.)